picard/ui/tablebaseddialog.py

```diff
@@ -175,20 +179,62 @@ def highlight_row(self, row):
             index = model.index(row, column)
             model.setData(index, highlight_brush, QtCore.Qt.ItemDataRole.BackgroundRole)
 
+    def header(self):
+        return self.table.horizontalHeader()
+
+    def restore_default_columns(self):
+        self.table.set_labels(_(c.title) for c in self.columns)
+
+        header = self.header()
+
+        def sort_indicator_changed(idx, order):
+            self._sort_column_index = idx
+            self._sort_order = order
+
+        header.sortIndicatorChanged.connect(sort_indicator_changed)
+
+        header.setStretchLastSection(True)
+        header.setDefaultAlignment(QtCore.Qt.AlignmentFlag.AlignLeft | QtCore.Qt.AlignmentFlag.AlignVCenter)
+        header.setDefaultSectionSize(self.columns.default_width)
+
+        for i, c in enumerate(self.columns):
+            # header.show_column(i, c.is_default)
+            if c.width is not None:
+                header.resizeSection(i, c.width)
+            if c.resizeable:
+                header.setSectionResizeMode(i, QtWidgets.QHeaderView.ResizeMode.Interactive)
+            else:
+                header.setSectionResizeMode(i, QtWidgets.QHeaderView.ResizeMode.Fixed)
+
     def prepare_table(self):
-        labels = tuple(_(c.title) for c in self.columns)
-        self.table.prepare(labels)
-        self.restore_table_header_state()
+        self.table.clear_contents()
+        # Disable sorting, as new elements will be added, and having sorting enabled
+        # doesn't play well.
+        # It will be eventually re-enabled in show_table()
+        self.table.setSortingEnabled(False)
 
     def show_table(self, sort_column=None, sort_order=QtCore.Qt.SortOrder.DescendingOrder):
         self.add_widget_to_center_layout(self.table)
-        self.table.horizontalHeader().setSortIndicatorShown(self.sorting_enabled)
+        header = self.header()
+        header.setSortIndicatorShown(self.sorting_enabled)
+        if self.sorting_enabled:
+            if self._sort_column_index is None or self._sort_order is None:
+                # Initialize the sort column & order based on passed parameters
+                if sort_column is None:
+                    pos = 0
+                else:
+                    pos = self.columns.pos(sort_column)
+                # It will trigger a call to sort_indicator_changed()
+                # This will set _sort_column_index and _sort_order to non-None values
+                header.setSortIndicator(pos, sort_order)
+        else:
+            # no indicator
+            # https://doc.qt.io/qt-6/qheaderview.html#setSortIndicator
+            header.setSortIndicator(-1, sort_order)
+
+        # Enabling sorting will sort using current sort colum & order
```

Comment:
Missing the 'n' in 'column'.

Reply:
Fixed in [a182160](https://github.com/metabrainz/picard/pull/2658/commits/a1821603d299813e231028d316ffd6f780b376a7)